src/com/xilinx/rapidwright/rwroute/RouteNode.java

```diff
@@ -0,0 +1,810 @@
+                        assert(length == 0);
+                        break;
+                    case NODE_LOCAL:
+                    case INTENT_DEFAULT:
+                    case NODE_LAGUNA_OUTPUT:
+                        assert(length <= 1);
+                        break;
+                    case NODE_SINGLE:
+                        assert(length <= 2);
+                        if (length > 1) baseCost *= length;
+                        break;
+                    case NODE_DOUBLE:
+                        if (endTileXCoordinate != node.getTile().getTileXCoordinate()) {
+                            baseCost = 0.4f * length;
+                        }
+                        break;
+                    case NODE_HQUAD:
+                        assert (length != 0 || node.getAllDownhillNodes().isEmpty());
+                        baseCost = 0.35f * length;
+                        break;
+                    case NODE_VQUAD:
+                        // In case of U-turn nodes
+                        if (length != 0) baseCost = 0.15f * length;// VQUADs have length 4 and 5
+                        break;
+                    case NODE_HLONG:
+                        assert (length != 0 || node.getAllDownhillNodes().isEmpty());
+                        baseCost = 0.15f * length;// HLONGs have length 6 and 7
+                        break;
+                    case NODE_VLONG:
+                        baseCost = 0.7f;
+                        break;
+                    default:
+                        throw new RuntimeException(ic.toString());
+                }
+                break;
+            case PINFEED_I:
+            case PINBOUNCE:
+                baseCost = 0.4f;
+                break;
+            case PINFEED_O:
+                baseCost = 1f;
+                break;
+            default:
+                throw new RuntimeException(type.toString());
+        }
+    }
+
+    /**
+     * Checks if a RouteNode Object has been used by more than one users.
+     * @return true, if a RouteNode Object has been used by multiple users.
+     */
+    public boolean isOverUsed() {
+        return RouteNode.capacity < getOccupancy();
+    }
+
+    public boolean willOverUse(NetWrapper netWrapper) {
+        int occ = getOccupancy();
+        return occ > RouteNode.capacity || (occ == RouteNode.capacity && countConnectionsOfUser(netWrapper) == 0);
+    }
+
+    /**
+     * Checks if a RouteNode Object has been used.
+     * @return true, if a RouteNode Object has been used.
+     */
+    public boolean isUsed() {
+        return getOccupancy() > 0;
+    }
+
+    /**
+     * Checks if a RouteNode Object are illegally driven by multiple drivers.
+     * @return true, if a RouteNode Object has multiple drivers.
+     */
+    public boolean hasMultiDrivers() {
+        return RouteNode.capacity < uniqueDriverCount();
+    }
+
+    public static final EnumSet<TileTypeEnum> lagunaTileTypes = EnumSet.of(
```

Comment:
Should this be moved to [`Utils.java`](https://github.com/Xilinx/RapidWright/blob/master/src/com/xilinx/rapidwright/util/Utils.java)?